README.md

```diff
@@ -149,82 +149,6 @@ if you run into any errors or problems with the prebuilt wheels.
 
 ### Building JAX from source
```

Comment:
This section header now seems misleading.

In addition, some users will need to build from source, so even if the build instructions move elsewhere, we would still want a link here I think.

Reply:
Done